### PR TITLE
[dashboard] cache get SupportedWorkspaceClasses

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Cache the supportedWorkspaceClasses on the client

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/dashboard/src/components/SelectWorkspaceClassComponent.tsx
Comment (easyCZ) [inline components/dashboard/src/components/SelectWorkspaceClassComponent.tsx:21]: Noob Q, why do we need the `memo` when we're already using the cache on `useWorkspaceClasses`?

**Review by easyCZ — APPROVED** [review]
LGTM

/hold if you want another review

File: components/dashboard/src/components/SelectWorkspaceClassComponent.tsx
Comment (svenefftinge) [reply components/dashboard/src/components/SelectWorkspaceClassComponent.tsx:21]: React runs the main function frequently (on every rerender) but would run what is in `useMemo` only if the dependency array has changed. So we basically cache the creation of the DropDownElement objects which included the WorkspaceClassDFropDownElement and its children. It's IMHO not really necessary and in many cases react users are prematurely optimizing here. This case is actually one of those places where it makes more sense.

Comment (svenefftinge) [reply components/dashboard/src/components/SelectWorkspaceClassComponent.tsx:21]: this caching using the dependency array works the same for `useEffect` and `useCallback` btw. They only get executed when the dependency array has changed.

**Pull request merged**
